bower install failed at dijit-themes

Hello,

I'm trying to run npm run build but hitting this error. Any idea on resolving this issue?

.... OUTPUT OMITTED ...
bower dojox#1.14.0                     progress received 12.2MB
bower dojox#v1.14.0/esri-3.26.0        resolved https://github.com/Esri/dojox.git#v1.14.0/esri-3.26.0
bower dojox#1.14.0                     progress received 13.1MB
bower dijit-themes#1.14.0             ENOTFOUND Package dijit-themes not found
child_process.js:645
    throw err;
    ^

Error: Command failed: bower install esri=arcgis-js-api#3.26 --force-latest  --config.directory=.
    at checkExecSyncError (child_process.js:602:13)

Thanks for the help!

Hi George, yes i did.

Here is the full log. Let me know if I'm missing anything? Thanks!

```
C:\Users\Llew\Documents\tfnsw-wab-widget>bower cache clean
bower deleted       Cached package put-selector: C:\Users\Llew\AppData\Local\bower\cache\packages\0642bfad576a2adb90fdc9ce748fdbc9\0.3.6
bower deleted       Cached package dstore: C:\Users\Llew\AppData\Local\bower\cache\packages\430f056772a34c4e6a1ba50f995d55d9\1.1.2
bower deleted       Cached package xstyle: C:\Users\Llew\AppData\Local\bower\cache\packages\5b929a5caa306d94f4d35faf3b72ab6d\0.3.2
bower deleted       Cached package dojo: C:\Users\Llew\AppData\Local\bower\cache\packages\014f5cfcdfe6ad61d53d4d042c94421e\1.14.0
bower deleted       Cached package dojo: C:\Users\Llew\AppData\Local\bower\cache\packages\014f5cfcdfe6ad61d53d4d042c94421e\1.14.2
bower deleted       Cached package dojo: C:\Users\Llew\AppData\Local\bower\cache\packages\20ebf9ca48823b72a3f0773052e7fe6e\v1.14.0%2Fesri-3.26.0
bower deleted       Cached package moment: C:\Users\Llew\AppData\Local\bower\cache\packages\bc513c8c04bcb090c1bc9c31b7ba3b0f\2.22.1
bower deleted       Cached package dgrid: C:\Users\Llew\AppData\Local\bower\cache\packages\0b814cd436207776181c38ffb9eda277\0.3.17%2Fesri-3.20.0
bower deleted       Cached package dgrid: C:\Users\Llew\AppData\Local\bower\cache\packages\0b814cd436207776181c38ffb9eda277\v1.2.1%2Fesri-3.22.0
bower deleted       Cached package dijit: C:\Users\Llew\AppData\Local\bower\cache\packages\aa9aaeb842cef54a90766ff53facce4f\1.14.0
bower deleted       Cached package dijit: C:\Users\Llew\AppData\Local\bower\cache\packages\e582207f69701f3908e1d7c1ce352b53\v1.14.0%2Fesri-3.26.0
bower deleted       Cached package dojox: C:\Users\Llew\AppData\Local\bower\cache\packages\b82cf372a5a6f33d12e1e3c9a0f5bcc3\v1.14.0%2Fesri-3.26.0
bower deleted       Cached package dojo-util: C:\Users\Llew\AppData\Local\bower\cache\packages\1623c8eefe0e4916583ed25715f90528\v1.14.0%2Fesri-3.26.0
bower deleted       Cached package arcgis-js-api: C:\Users\Llew\AppData\Local\bower\cache\packages\60e815d1ef65689c579904a919a10851\3.26.1

C:\Users\Llew\Documents\tfnsw-wab-widget>npm run build

> TfNSW@1.0.0 build C:\Users\Llew\Documents\tfnsw-wab-widget
> esri-wab-build C:/Users/Llew/Documents/WebAppBuilderForArcGIS/server/apps/2

Cleaning dist directory...
########## BUILD START TIME: Fri Nov 09 2018 07:42:39 GMT+0800 (Malay Peninsula Standard Time) ##########
bower esri#3.26             not-cached https://github.com/Esri/arcgis-js-api.git#3.26
bower esri#3.26                resolve https://github.com/Esri/arcgis-js-api.git#3.26
bower esri#3.26               download https://github.com/Esri/arcgis-js-api/archive/3.26.1.tar.gz
bower esri#3.26                extract archive.tar.gz
bower esri#3.26               resolved https://github.com/Esri/arcgis-js-api.git#3.26.1
bower dojo#v1.14.0/esri-3.26.0       not-cached https://github.com/Esri/dojo.git#v1.14.0/esri-3.26.0
bower dojo#v1.14.0/esri-3.26.0          resolve https://github.com/Esri/dojo.git#v1.14.0/esri-3.26.0
bower dojox#v1.14.0/esri-3.26.0      not-cached https://github.com/Esri/dojox.git#v1.14.0/esri-3.26.0
bower dojox#v1.14.0/esri-3.26.0         resolve https://github.com/Esri/dojox.git#v1.14.0/esri-3.26.0
bower dijit#v1.14.0/esri-3.26.0      not-cached https://github.com/Esri/dijit.git#v1.14.0/esri-3.26.0
bower dijit#v1.14.0/esri-3.26.0         resolve https://github.com/Esri/dijit.git#v1.14.0/esri-3.26.0
bower util#v1.14.0/esri-3.26.0       not-cached https://github.com/Esri/dojo-util.git#v1.14.0/esri-3.26.0
bower util#v1.14.0/esri-3.26.0          resolve https://github.com/Esri/dojo-util.git#v1.14.0/esri-3.26.0
bower dgrid#0.3.17/esri-3.20.0       not-cached https://github.com/Esri/dgrid.git#0.3.17/esri-3.20.0
bower dgrid#0.3.17/esri-3.20.0          resolve https://github.com/Esri/dgrid.git#0.3.17/esri-3.20.0
bower dgrid1#v1.2.1/esri-3.22.0      not-cached https://github.com/Esri/dgrid.git#v1.2.1/esri-3.22.0
bower dgrid1#v1.2.1/esri-3.22.0         resolve https://github.com/Esri/dgrid.git#v1.2.1/esri-3.22.0
bower put-selector#0.3.6             not-cached https://github.com/kriszyp/put-selector.git#0.3.6
bower put-selector#0.3.6                resolve https://github.com/kriszyp/put-selector.git#0.3.6
bower dstore#1.1.2                   not-cached https://github.com/SitePen/dstore.git#1.1.2
bower dstore#1.1.2                      resolve https://github.com/SitePen/dstore.git#1.1.2
bower moment#2.22.1                  not-cached https://github.com/moment/moment.git#2.22.1
bower moment#2.22.1                     resolve https://github.com/moment/moment.git#2.22.1
bower xstyle#0.3.2                   not-cached https://github.com/kriszyp/xstyle.git#0.3.2
bower xstyle#0.3.2                      resolve https://github.com/kriszyp/xstyle.git#0.3.2
bower dgrid#0.3.17/esri-3.20.0         download https://github.com/Esri/dgrid/archive/0.3.17/esri-3.20.0.tar.gz
bower dgrid1#v1.2.1/esri-3.22.0        download https://github.com/Esri/dgrid/archive/v1.2.1/esri-3.22.0.tar.gz
bower dijit#v1.14.0/esri-3.26.0        download https://github.com/Esri/dijit/archive/v1.14.0/esri-3.26.0.tar.gz
bower util#v1.14.0/esri-3.26.0         download https://github.com/Esri/dojo-util/archive/v1.14.0/esri-3.26.0.tar.gz
bower dojo#v1.14.0/esri-3.26.0         download https://github.com/Esri/dojo/archive/v1.14.0/esri-3.26.0.tar.gz
bower dojox#v1.14.0/esri-3.26.0        download https://github.com/Esri/dojox/archive/v1.14.0/esri-3.26.0.tar.gz
bower put-selector#0.3.6               download https://github.com/kriszyp/put-selector/archive/v0.3.6.tar.gz
bower xstyle#0.3.2                     download https://github.com/kriszyp/xstyle/archive/v0.3.2.tar.gz
bower dstore#1.1.2                     download https://github.com/SitePen/dstore/archive/v1.1.2.tar.gz
bower moment#2.22.1                    download https://github.com/moment/moment/archive/2.22.1.tar.gz
bower dgrid#0.3.17/esri-3.20.0          extract archive.tar.gz
bower dgrid1#v1.2.1/esri-3.22.0         extract archive.tar.gz
bower dgrid#0.3.17/esri-3.20.0         resolved https://github.com/Esri/dgrid.git#0.3.17/esri-3.20.0
bower put-selector#0.3.6                extract archive.tar.gz
bower put-selector#0.3.6               resolved https://github.com/kriszyp/put-selector.git#0.3.6
bower dijit#v1.14.0/esri-3.26.0         extract archive.tar.gz
bower dojo#v1.14.0/esri-3.26.0          extract archive.tar.gz
bower dstore#1.1.2                      extract archive.tar.gz
bower dgrid1#v1.2.1/esri-3.22.0        resolved https://github.com/Esri/dgrid.git#v1.2.1/esri-3.22.0
bower dojo#>=1.8.9                   not-cached https://github.com/dojo/dojo.git#>=1.8.9
bower dojo#>=1.8.9                      resolve https://github.com/dojo/dojo.git#>=1.8.9
bower dstore#1.1.2                     resolved https://github.com/SitePen/dstore.git#1.1.2
bower dojo#>=1.8.1                   not-cached https://github.com/dojo/dojo.git#>=1.8.1
bower dojo#>=1.8.1                      resolve https://github.com/dojo/dojo.git#>=1.8.1
bower xstyle#0.3.2                      extract archive.tar.gz
bower xstyle#0.3.2                     resolved https://github.com/kriszyp/xstyle.git#0.3.2
bower dojo#>=1.8.1                     download https://github.com/dojo/dojo/archive/1.14.2.tar.gz
bower dojo#>=1.8.9                     download https://github.com/dojo/dojo/archive/1.14.2.tar.gz
bower moment#2.22.1                     extract archive.tar.gz
bower dojo#v1.14.0/esri-3.26.0         resolved https://github.com/Esri/dojo.git#v1.14.0/esri-3.26.0
bower moment#2.22.1                    resolved https://github.com/moment/moment.git#2.22.1
bower dijit#v1.14.0/esri-3.26.0        resolved https://github.com/Esri/dijit.git#v1.14.0/esri-3.26.0
bower dojox#v1.14.0/esri-3.26.0         extract archive.tar.gz
bower dojo#>=1.8.9                      extract archive.tar.gz
bower dojo#>=1.8.1                      extract archive.tar.gz
bower util#v1.14.0/esri-3.26.0         progress received 7.5MB
bower util#v1.14.0/esri-3.26.0         progress received 8.8MB
bower dojo#>=1.8.9                     resolved https://github.com/dojo/dojo.git#1.14.2
bower util#v1.14.0/esri-3.26.0         progress received 10.0MB
bower dojo#>=1.8.1                     resolved https://github.com/dojo/dojo.git#1.14.2
bower dojo#1.14.0                    not-cached https://github.com/dojo/dojo.git#1.14.0
bower dojo#1.14.0                       resolve https://github.com/dojo/dojo.git#1.14.0
bower dojo#1.14.0                      download https://github.com/dojo/dojo/archive/1.14.0.tar.gz
bower util#v1.14.0/esri-3.26.0         progress received 11.4MB
bower util#v1.14.0/esri-3.26.0         progress received 13.0MB
bower util#v1.14.0/esri-3.26.0         progress received 14.9MB
bower dojo#1.14.0                       extract archive.tar.gz
bower util#v1.14.0/esri-3.26.0         progress received 17.1MB
bower util#v1.14.0/esri-3.26.0         progress received 19.6MB
bower util#v1.14.0/esri-3.26.0         progress received 22.1MB
bower dojo#1.14.0                      resolved https://github.com/dojo/dojo.git#1.14.0
bower util#v1.14.0/esri-3.26.0          extract archive.tar.gz
bower util#v1.14.0/esri-3.26.0         resolved https://github.com/Esri/dojo-util.git#v1.14.0/esri-3.26.0
bower dojox#1.14.0                   not-cached https://github.com/dojo/dojox.git#1.14.0
bower dojox#1.14.0                      resolve https://github.com/dojo/dojox.git#1.14.0
bower dijit#1.14.0                   not-cached https://github.com/dojo/dijit.git#1.14.0
bower dijit#1.14.0                      resolve https://github.com/dojo/dijit.git#1.14.0
bower dojox#1.14.0                     download https://github.com/dojo/dojox/archive/1.14.0.tar.gz
bower dijit#1.14.0                     download https://github.com/dojo/dijit/archive/1.14.0.tar.gz
bower dijit#1.14.0                      extract archive.tar.gz
bower dojox#v1.14.0/esri-3.26.0        resolved https://github.com/Esri/dojox.git#v1.14.0/esri-3.26.0
bower dojox#1.14.0                      extract archive.tar.gz
bower dijit#1.14.0                     resolved https://github.com/dojo/dijit.git#1.14.0
bower dijit-themes#1.14.0             ENOTFOUND Package dijit-themes not found
child_process.js:645
    throw err;
    ^

Error: Command failed: bower install esri=arcgis-js-api#3.26 --force-latest  --config.directory=.
    at checkExecSyncError (child_process.js:602:13)
    at execSync (child_process.js:642:13)
    at Object.exports.build (C:\Users\Llew\Documents\tfnsw-wab-widget\node_modules\esri-wab-build\app\buildTool.js:56:3)
    at Object.exports.buildApp (C:\Users\Llew\Documents\tfnsw-wab-widget\node_modules\esri-wab-build\app\remoteBuild.js:20:13)
    at Object.<anonymous> (C:\Users\Llew\Documents\tfnsw-wab-widget\node_modules\esri-wab-build\app\build:12:16)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! TfNSW@1.0.0 build: `esri-wab-build C:/Users/Llew/Documents/WebAppBuilderForArcGIS/server/apps/2`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the TfNSW@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Llew\AppData\Roaming\npm-cache\_logs\2018-11-08T23_43_54_162Z-debug.log
```

--- a/app/buildTool.js
+++ b/app/buildTool.js
@@ -53,10 +53,19 @@ exports.build = function(buildPath) {
     return false;
   });
   const apiVersion = envJs.tokens[apiVersionIndex + 2].value;
+
+  // We always include the arcgis-js-api dependency. If the JS API version is greater than 3.25, we must include dijit-themes:
+  // https://github.com/Esri/jsapi-resources/commit/7f26c7bc7a1ee305102cd7b1f95d1631df0edbd5#diff-265400d6fce2c9b60ecb6dbea36d979f
+  let dependencies = [`esri=arcgis-js-api#${apiVersion}`];
+  if (apiVersion > 3.25) {
+    dependencies.push(
+      "dijit-themes=https://github.com/dojo/dijit-themes.git#1.14.0"
+    );
+  }
   execSync(
-    "bower install esri=arcgis-js-api#" +
-      apiVersion +
-      " --force-latest  --config.directory=.",
+    `bower install ${dependencies.join(
+      " "
+    )} --force-latest  --config.directory=.`,
     syncArgs
   );
 
@@ -146,7 +155,6 @@ exports.build = function(buildPath) {
           outputPath: outputPath,
           outputZipPath: outputZipPath
         });
-
       } catch (err) {
         console.log("Oh no! There was an error zipping the final build.", err);
         reject(err);

